code/modules/client/client_procs.dm

```diff
@@ -898,3 +901,11 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 		screen -= S
 		qdel(S)
 	char_render_holders = null
+
+/client/proc/toggle_popup_menus(mode = TRUE, disable = FALSE, forced = FALSE)
+	if(prefs.context_menu_on && !forced)
+		return
+	if(mob.combatmode)
+		mode = FALSE
+	show_popup_menus = mode
+	prefs.context_menu_on = disable
```

Comment:
What is the point of having every single toggle_popup_menu() call directly change the client's preferences?